internal/bowtie/resources/dns_block_list.go

```diff
@@ -0,0 +1,218 @@
+package resources
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	"github.com/bowtieworks/terraform-provider-bowtie/internal/bowtie/client"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ resource.Resource = &dnsBlockListResource{}
+var _ resource.ResourceWithImportState = &dnsBlockListResource{}
+
+type dnsBlockListResource struct {
+	client *client.Client
+}
+
+type dnsBlockListResourceModel struct {
+	ID              types.String `tfsdk:"id"`
+	Name            types.String `tfsdk:"name"`
+	LastUpdated     types.String `tfsdk:"last_updated"`
+	Upstream        types.String `tfsdk:"upstream"`
+	OverrideToAllow types.String `tfsdk:"override_to_allow"`
+}
+
+func NewDNSBlockListResource() resource.Resource {
+	return &dnsBlockListResource{}
+}
+
+func (bl *dnsBlockListResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_dns_block_list"
+}
+
+type urlValidator struct{}
```

Comment:
I thought at one point I had found some common usecases like this and cidrs but can't remember where

Reply:
Yeah, I thought this'd be something that I could pick up from elsewhere but it wasn't hard to just implement.